### PR TITLE
Changed win logic

### PR DESCRIPTION
**Pull request by jaydenwindle**
Fixes:
- Added a `winners` list to the giveaway model so we can persist the list of winners
- Added `item_won` to user model to store the product id if the user wins
- Changed `show` route to list participants if giveaway is still ongoing, otherwise list all the winners and the things they won
- Disabled the ajax participant count update in `dash_main.js` because it was running too often and causing shopify to reject subsequent API calls

### CODE REVIEW

File: app/controllers/giveaways_controller.rb [hunk 1]
```diff
@@ -97,27 +105,12 @@ def create
...
 
-      final = {}
-      winners.keys.each do |winner_id|
-        winner = giveaway.users.find_by(id: winner_id)
-        product = giveaway.products.find_by(id: winners[winner_id])
-        final[winner[:name]] = { name:  product.name, email: winner.email, coupon: coupons[product.id]}
-      end
+      giveaway.choose_winners
+      giveaway.isActive = false
+      giveaway.save
```
Comment (vikrambombhi) [inline app/controllers/giveaways_controller.rb:111]: can we use ```giveaway.update(:isActive => true)``` for small changes like this?

File: app/models/giveaway.rb [hunk 2]
```diff
@@ -1,16 +1,23 @@
 class Giveaway < ApplicationRecord
     has_many :users
+    has_many :winners, :class_name => "User" 
     has_many :products
   
     def choose_winners   
-      winners = {} 
       all_applicants = users.to_a.shuffle
+      p all_applicants
```
Comment (vikrambombhi) [inline app/models/giveaway.rb:8]: im seeing alot of these prints everywhere. They are fine for testing but I think we should remove them for master

**Review by vikrambombhi — COMMENTED** [review]
Other than some small things this looks great. Ill be honest im not 100% sure how show is working but I trust you on that

File: app/models/giveaway.rb (hunk 2 above)
Comment (jaydenwindle) [reply app/models/giveaway.rb:8]: Good call. I'll get rid of those.

File: app/controllers/giveaways_controller.rb (hunk 1 above)
Comment (jaydenwindle) [reply app/controllers/giveaways_controller.rb:111]: Yep, can do!

**Pull request closed without merging**